pkg/controller/plan/kubevirt.go

```diff
@@ -1166,6 +1175,91 @@ func (r *KubeVirt) virtualMachine(vm *plan.VMStatus) (object *cnv.VirtualMachine
 	return
 }
 
+// Attempt to find a suitable preference.
+func (r *KubeVirt) vmPreference(vm *plan.VMStatus) (virtualMachine *cnv.VirtualMachine, err error) {
+	config, err := r.getOsMapConfig(r.Source.Provider.Type())
+	if err != nil {
+		return
+	}
+	preferenceName, err := r.Builder.PreferenceName(vm.Ref, config)
+	if err != nil {
+		return
+	}
+
+	preferenceName, kind, err := r.getPreference(vm, preferenceName)
+	if err != nil {
+		return
+	}
+
+	virtualMachine = r.emptyVm(vm)
+	virtualMachine.Spec.Preference = &cnv.PreferenceMatcher{Name: preferenceName, Kind: kind}
+	return
+}
+
+func (r *KubeVirt) getOsMapConfig(providerType v1beta1.ProviderType) (configMap *core.ConfigMap, err error) {
+	list := &core.ConfigMapList{}
+	label := map[string]string{}
+	switch providerType {
+	case api.VSphere:
+		label["subject"] = "vsphere-osmap-config"
+	case api.OVirt:
+		label["subject"] = "ovirt-osmap-config"
+	default:
+		return
+	}
+	err = r.Destination.Client.List(
```

Comment:
this part is tricky... it means that the configmaps on destination cluster, which we can't promise.
the workaround for it is to post the configmaps to the destination cluster, either by using our deployed ones or by hardcode it and post it...